packages/inertia/src/inertia.js

```diff
@@ -68,7 +68,16 @@ export default {
     return this.visitId
   },
 
-  visit(url, { method = 'get', data = {}, replace = false, preserveScroll = false, preserveState = false, only = [], headers = {}} = {}) {
+  visit(url, { method = 'get', data = {}, replace = false, preserveScroll = false, preserveState = false, only = [], headers = {} } = {}) {
+    let event = new CustomEvent('inertia:start', {
+      cancelable: true,
+      detail: { url: url, ...arguments[1] }
+    })
+
+    if (!document.dispatchEvent(event)) {
+      return
```

Comment:
Not sure if this can really be considered a bug, but, returning void causes an inconsistent return type.

Example: PingCRM's login page chains a `then` on the end of `this.$inertia.post(..)`, which (when interrupted by an event listener) causes the following error:
```js
"TypeError: Cannot read property 'then' of undefined"
```

```suggestion
      return Promise.reject()
```

With this, it'd still fail the `.then`, but at least it would consistently return a promise.